compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -963,6 +965,19 @@ public void visit(BLangArrayType arrayType) {
 
     @Override
     public void visit(BLangUserDefinedType userDefinedType) {
+        BType type = userDefinedType.getBType();
+        if (type == null) {
+            result = userDefinedType;
+            return;
+        }
+        if (type.tag == TypeTags.RECORD) {
+            ((BRecordType) type).enclMapSymbols = collectClosureMapSymbols(env, false);
```

Comment:
So `enclMapSymbols` are relevant to the type?

Reply:
Yes. This is used to maintain the references of variables within the 'record' type node.